### PR TITLE
Fix race condition and remove ethereum_mock dependency

### PR DESCRIPTION
**Pull request by tudor-malene**
major changes:

1. Remove dependency on ethereum_mock.
2. Submit both the current L1 block and the parent on the RPC call to the l2 node.

### CODE REVIEW

File: go/obscuronode/enclave/enclave.go [hunk 1]
```diff
@@ -176,7 +192,10 @@ func (e *enclaveImpl) SubmitRollup(rollup common2.ExtRollup) {
 		Header:       rollup.Header,
 		Transactions: decryptTransactions(rollup.Txs),
 	}
-	e.db.StoreRollup(&r)
+	// only store if the parent exists
+	if e.db.ExistRollup(r.Header.ParentHash) {
```
Comment (otherview) [inline go/obscuronode/enclave/enclave.go:196]: Should we log when it's trying to submit a rollup that the parent doesn't exist ?

File: go/obscuronode/node.go [hunk 2]
```diff
@@ -122,7 +121,9 @@ func sendInterrupt(interrupt *int32) *int32 {
 func (a *Node) processBlocks(blocks []common.EncodedBlock, interrupt *int32) {
 	var result enclave.SubmitBlockResponse
 	for _, block := range blocks {
-		result = a.Enclave.SubmitBlock(block.DecodeBlock().ToExtBlock())
+		if block != nil {
```
Comment (otherview) [inline go/obscuronode/node.go:124]: Just out of curiosity, why would these blocks ever be nil ?

**Review by otherview — COMMENTED** [review]
Looks good, I would suggest adding a log, running `gofumpt -l -w .` to fix some small lint issues and `golangci-lint run` can be used to check if the linting is passing.

File: go/obscuronode/node.go (hunk 2 above)
Comment (tudor-malene) [reply go/obscuronode/node.go:124]: good point, I should add a comment.
it's somewhat ugly. On the genesis block , the parent is null

File: go/obscuronode/enclave/enclave.go (hunk 1 above)
Comment (tudor-malene) [reply go/obscuronode/enclave/enclave.go:195]: I'll add something. Might help with invesitaging.
This scenario is basically when a node receives a rollup during the gossip period, and it's most likely behind.

**Review by otherview — APPROVED** [review]

**Pull request merged**